### DISCUSSION TITLE
Using caching on tags

### DISCUSSION THREAD

**Discussion by martinmiglio** [actions/cache#1178] [Q&A]
I have a lengthy build process in my release workflow. I run this workflow on new version tags, and normally (in a local build environment) I would be able to cache some of the build artifacts to speed up the next build, but it appears the caches aren't available cross-tag. 

In [the documentation,](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache) it states that caches are shared from parent branches to their children branches. It doesn't seem like this works with tags as they don't really have parents (but they are associated with a commit which does have an associated branch).

Is there some kinda of solution for this? Is there a reason there isn't an option to allow cross-branch/tag caches? Without this, the cache action feels kind of useless except for testing workflows which run on every branch.

There is discussion similar to this in #1175.

**Comment by arxeiss** [comment]
Have you figured out more info? I'm running some code on release.
And it seems that the cache created during that flow is not available in branch/PRs either.

```yaml
on:
    release:
        types:
            - published
```
